README.md

```diff
@@ -10,31 +10,32 @@ Read the [Notes and FAQ](#notes-and-faq)
 
 ```rust
 use worker::*;
+use serde::{Serialize, Deserialize};
 
-#[event(fetch)]
-pub async fn main(req: Request, env: Env, _ctx: worker::Context) -> Result<Response> {
-    console_log!(
-        "{} {}, located at: {:?}, within: {}",
-        req.method().to_string(),
-        req.path(),
-        req.cf().coordinates().unwrap_or_default(),
-        req.cf().region().unwrap_or("unknown region".into())
-    );
-
-    if !matches!(req.method(), Method::Post) {
-        return Response::error("Method Not Allowed", 405);
-    }
 
-    if let Some(file) = req.form_data().await?.get("file") {
-        return match file {
-            FormEntry::File(buf) => {
-                Response::ok(&format!("size = {}", buf.bytes().await?.len()))
-            }
-            _ => Response::error("`file` part of POST form must be a file", 400),
-        };
-    }
+#[derive(Serialize, Deserialize)]
+struct Thing {
+	thing_id: String,
+	desc: String,
+	num: u32,
+}
 
-    Response::error("Bad Request", 400)
+#[event(fetch, respond_with_errors)]
+pub async fn main(request: Request, env: Env, _ctx: Context) -> Result<Response> {
+	Router::new()
+		.get_async("/:id", |_, ctx| async move {
+			let id = ctx.param("id").unwrap();
+			let d1 = ctx.env.d1("things-db")?;
+			let statement = d1.prepare("SELECT * FROM things WHERE thing_id = ?1");
+			let query = statement.bind(&[id.into()])?;
+			let result = query.first::<Thing>(None).await?;
```

Comment:
nit: I find these easier to read as `let result: Option<Thing> = query.first(Nonte).await?;` Since the turbo-fish with arg as parameter. Also, I think `.first()` is just a convenience wrapper around `.all()` so I think you might also need to add a `LIMIT 1` to your sql query. I'm not sure where the d1 source lives but I did find the deprecated dep on npm: https://www.npmjs.com/package/@cloudflare/d1?activeTab=code

Reply:
thanks.
Your suggestion looks better, fix it.